platform/src/common/components/Charts/MoreInsightsChart.jsx

```
@@ -100,26 +100,56 @@ const MoreInsightsChart = React.memo(
+        setDataError('Data is not an array.');
+        return [];
+      }
+
+      return data.map((item, index) => {
+        const newItem = { ...item };
+        if (!newItem.time) {
+          console.warn(`Data item at index ${index} is missing 'time' field.`);
+          newItem.time = 'N/A';
+        } else {
+          const date = new Date(newItem.time);
+          if (isNaN(date.getTime())) {
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)